packages/binding/src/services/diagnostics/validators/check-required.ts

```diff
@@ -1,24 +1,32 @@
-import { findRange } from "../../../utils";
+import { findRange, getAltTypesPrime } from "../../../utils";
 import {
   BindingIssue,
   BINDING_ISSUE_TYPE,
   BindingInfoElement,
 } from "../../../types";
 import { BindingParserTypes as BindingTypes } from "@ui5-language-assistant/binding-parser";
 import { t } from "../../../i18n";
+import { UI5Aggregation } from "@ui5-language-assistant/semantic-model-types";
 
 export const checkRequiredElement = (
   element: BindingTypes.StructureValue,
-  bindingElements: BindingInfoElement[]
+  bindingElements: BindingInfoElement[],
+  aggregation: UI5Aggregation | undefined
 ): BindingIssue[] => {
   // check required element
   const reqEl = bindingElements.find((i) => i.required);
   if (reqEl) {
     // check required element is applied
-    const usedRequiredEl = element.elements.find(
+    let usedRequiredEl = element.elements.find(
       /* istanbul ignore next */
       (i) => i.key?.text === reqEl.name
     );
+    const altTypes = getAltTypesPrime(aggregation);
+    if (!usedRequiredEl && altTypes) {
+      // some property e.g `tooltip` can be used with both `aggregation binding info` or `property binding info`. Therefore `altTypes` is defined in design time.
+      // if `altTypes` is present, check if any element is used. This is a very broad check to avoid false diagnostic.
+      usedRequiredEl = element.elements[0];
```

Comment:
Does elements array always contain at least one element?
Maybe better here just exit the validation function with empty result?

Reply:
the exception is to check if there is at least one element present from either `aggregation property binding` or `property binding info`. If that element is not present, we still want to show diagnostic for mandatory property.